verifiers/internal/gha/slsaprovenance/v0.2/provenance.go

```diff
@@ -52,22 +38,19 @@ func New(payload []byte) (iface.Provenance, error) {
 	}
 
 	switch {
-	case a.Predicate.BuildType == byobDelegatorBuildType:
-		return &provenanceV02{
-			upperEnv: true,
-			prov:     a,
-		}, nil
-	case a.Predicate.BuildType == goBuilderBuildType ||
-		a.Predicate.BuildType == genericGeneratorBuildType ||
-		a.Predicate.BuildType == containerGeneratorBuildType ||
-		a.Predicate.BuildType == npmCLIBuildType ||
-		a.Predicate.BuildType == legacyBuilderBuildType ||
-		a.Predicate.BuildType == legacyGoBuilderBuildType:
+	case a.Predicate.BuildType == common.BYOBBuildTypeV0:
```

Comment:
Same comment as above, ie the caller should determine what build IDs are acceptable.

Reply:
In the part of the code I think the problem could be more serious. If say, another builder is compromised, it could replace the builder ID in the provenance to be of BYOB type, and replace the builder ID with a builder of its choice.

Wdut?

Reply:
Added a map of builder ID -> buildType to validate buildTypes.